### DISCUSSION TITLE
tailwind css responsive not working with any class like md:flex xl:block,..

### DISCUSSION THREAD

**Discussion by devntv** [tailwindlabs/tailwindcss#4457] [Help] [answered]
hello everyone!. Have a nice day
I have a problem when using Tailwinds as follows:
- when i responsive it still works with border, color, text-sm, background,... (md:text-sm, xl:bg-white,...)
but when i want responsive (with display) on devices, it not working with code md:flex, md:block, xl:grid,...

` <div className=" w-3/4 relative -ml-10 -mr-5 bg-cover h-auto hidden md:block xl:bg-white">work </div>`
- md:block not working
 - xl:bg-white works
 
 i build tailwind css + react

**Comment by therealgilles** [comment, marked as answer]
@devntv: did you maybe override 'display' in your tailwind config?

PS: If you want a precise and correct answer to your question, you need to give people enough information to be able to help you.

**Comment by simonswiss** [comment]
Hey @devntv 👋

Are you able to share a code reproduction of this problem, so we can take a look? It's practically impossible to help without this, since a class like `md:block` _does_ work by default in Tailwind CSS.

Happy to take a look if you can share something 👍

As @therealgilles, I'd look first into the config, checking if you didn't accidentally disable the `responsive` variant for the `display` core plugin.

**Comment by devntv** [comment]
Sorry for the delay.
thanks so much @therealgilles  @simonswiss .

Exactly, maybe I installed the wrong file tailwind.config.js, I forgot it must have `extends` in `variants`. But after i add this code in tailwind.config.js
`variants: { extends:{ display: ['group-hover', 'responsive','hover'], } }`
i noob, can help me?	
-display - it worked.
-but group-hover not working 

`group-hover:flex` not working, i want when i hover it will be show `<p>Hover</p>`  with my html as follow:
code: 
```
<div className="relative group object-cover">
      <img className="h-36 w-auto object-cover bg-fixed rounded overflow-hidden bg-center bg-contain sm-res:h-80" />
          <div className="absolute bottom-0 left-0 bg-gray-200 z-10 w-full justify-center items-center h-full bg-black-faded hidden group-hover:flex">
              <p className="flex items-center text-white font-bold mr-4 border-red-light">Hover</p>
         </div>
</div>

**Reply by therealgilles** [reply]
It should 'extend' instead of 'extends'.

**Reply by devntv** [reply]
yepp awesome,i saw, I didn't check the tailwind.config.js file carefully. thanks so much @therealgilles
